lab_4_retrieval_w_clustering/main.py

```diff
@@ -316,6 +506,22 @@ def get_clusters_info(self, num_examples: int) -> list[dict[str, int | list[str]
         Returns:
             list[dict[str, int| list[str]]]: List with information about each cluster
         """
+        if not isinstance(num_examples, int) or not num_examples or num_examples <= 0:
+            raise ValueError("Incorrect input with number")
```

Comment:
of course it is optional, but lets be programmers who respect users, just explain the problem

Reply:
fixed